File: PlusMinus.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'plusMinus' function below.
#
# The function accepts INTEGER_ARRAY arr as parameter.
#

#!/bin/python3

def plusMinus(arr):
    pos = 0
    neg = 0
    zero = 0

    for i in range(len(arr)):  
        if arr[i] > 0:
            pos += 1
        elif arr[i] < 0:  
            neg += 1
        else:
            zero += 1

    rpos = pos / len(arr)
    rneg = neg / len(arr)
    rz = zero / len(arr)

    print(f"{rpos:.6f}")
    print(f"{rneg:.6f}")
    print(f"{rz:.6f}")


if __name__ == '__main__':
    n = int(input().strip())
    arr = list(map(int, input().rstrip().split()))
    plusMinus(arr)

File: PlusMinus.php
<?php

/*
 * Complete the 'plusMinus' function below.
 *
 * The function accepts INTEGER_ARRAY arr as parameter.
 */

function plusMinus($arr) {
    // Write your code here
    $n = count($arr);
    $pos = 0;
    $neg = 0;
    $zero = 0;
    for($i = 0 ; $i < $n ; $i++){
        if($arr[$i]>0){
            $pos+=1;
        }
        if($arr[$i]<0){
            $neg+=1;
        }
        if($arr[$i]==0){
            $zero+=1;
        }
    }
    $proportionsPos = $pos/$n;
    $proportionsNeg = $neg/$n;
    $proportionsZ = $zero/$n; 
    
    echo number_format($proportionsPos,6) . "\n";
    echo number_format($proportionsNeg,6) . "\n";
    echo number_format($proportionsZ,6);
}

$n = intval(trim(fgets(STDIN)));

$arr_temp = rtrim(fgets(STDIN));

$arr = array_map('intval', preg_split('/ /', $arr_temp, -1, PREG_SPLIT_NO_EMPTY));

plusMinus($arr);
